src/reactor.rs

```diff
@@ -701,54 +685,48 @@ impl<H: Handler, P: Poll> Runtime<H, P> {
-                    .expect("listener index content doesn't match registered listeners");
-                self.poller.unregister(&listener);
+                log::debug!(target: "reactor", "Handling over listener {id}");
                 self.service.handover_listener(listener);
             }
             Action::UnregisterTransport(id) => {
-                let transport = self.transports.remove(&id).ok_or(Error::TransportUnknown(id))?;
-                let fd = transport.as_raw_fd();
-
+                let Some(transport) = self.unregister_transport(id) else {
+                    return Ok(())
+                };
                 #[cfg(feature = "log")]
-                log::debug!(target: "reactor", "Handling over transport {id} (fd={fd})");
-
-                self.transport_map
-                    .remove(&fd)
-                    .expect("transport index content doesn't match registered transports");
-                self.poller.unregister(&transport);
+                log::debug!(target: "reactor", "Handling over transport {id}");
                 self.service.handover_transport(transport);
             }
             Action::Send(id, data) => {
                 #[cfg(feature = "log")]
                 log::trace!(target: "reactor", "Sending {} bytes to {id}", data.len());
 
-                let transport = self.transports.get_mut(&id).ok_or_else(|| {
+                let Some(transport) = self.transports.get_mut(&id) else {
                     #[cfg(feature = "log")]
                     log::error!(target: "reactor", "Transport {id} is not in the reactor");
 
-                    Error::TransportUnknown(id)
-                })?;
-                transport.write_atomic(&data).map_err(|err| match err {
-                    WriteError::NotReady => {
+                    return Ok(())
+                };
+                match transport.write_atomic(&data) {
+                    Err(WriteError::NotReady) => {
                         #[cfg(feature = "log")]
                         log::error!(target: "reactor", internal = true; 
                                 "An attempt to write to transport {id} before it got ready");
-                        Error::WriteLogicError(id, data)
+                        panic!(
+                            "application business logic error: write to transport {id} which is \
+                             read-only or not ready for a write operation"
+                        );
                     }
-                    WriteError::Io(e) => {
+                    Err(WriteError::Io(e)) => {
                         #[cfg(feature = "log")]
-                        log::error!(target: "reactor", "Error writing to transport {id}: {e:?}");
-                        Error::WriteFailure(id, e)
+                        log::error!(target: "reactor", "Fatal error writing to transport {id}, disconnecting. Error details: {e:?}");
+                        self.unregister_transport(id);
```

Comment:
Do I receive an error when this happens though? I'd expect a `TransportDisconnect` error probably, and *not* a call to `handover_transport` without error.

Reply:
Updated in the pushed commit. I was under an impression there is no need to handover the transport (which is a part of the error message) to the service. And w/o handover there were no need in removing the previous version of the error.